### PR TITLE
[SRVCOM-1846]: Remove outdated module and add content to func getting started

### PR DESCRIPTION
**Pull request by abrennan89**
Version(s):
OCP 4.6+
OSD 4

Issue:
https://issues.redhat.com/browse/SRVCOM-1846

Link to docs preview:
- http://file.rdu.redhat.com/abrennan/250522/SRVCOM-1846/serverless/functions/serverless-functions-getting-started.html
- http://file.rdu.redhat.com/abrennan/250522/SRVCOM-1846/serverless/cli_tools/kn-func-ref.html

(build and deploy sections)

### CODE REVIEW

File: serverless/functions/serverless-functions-getting-started.adoc
Comment (lance) [inline serverless/functions/serverless-functions-getting-started.adoc:9]: I guess the only thing I would say is that this is a big "if". It's entirely likely that _most_ OpenShift administrators haven't exposed the internal registry in this way - especially not for junior or mid level developers, who could easily be the kind of developers writing functions. If this is the case, and they try to deploy a function without the proper privileges, they will get an error. I think it might be important to add some cautionary text, e.g.

> **Warning**
> If you are logged in to an OpenShift cluster and do not have authorization for the Container Registry, you will need to configure your default image registry using the `FUNC_REGISTRY` environment variable.

Or something to that effect but better. ;)

Comment (matejvasek) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: The registry does not need to be exposed to outer internet and it is used by default now (for OpenShift clusters).

>Usually this is a public registry, such as Quay. However, you can use the integrated OpenShift Container Registry instead if it has been exposed by a cluster administrator.

Should be more like:
>By default the internal OpenShift registry is used,  this behariour can be overridden by the `-r` flag.

Comment (lance) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: > The registry does not need to be exposed to outer internet and it is used by default now (for OpenShift clusters).

Hm OK. I feel like I have run into errors before if I'm logged into an openshift registry. Or maybe it's just picking up an old kubecontext for a cluster I don't have around anymore. 🤷‍♂️

Comment (matejvasek) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: I think @rhuss onece saw such a failure. 

I tested this with `developer` account on CRC and in non-admin in Sandbox and it works.
Also @jrangelramos  tested this with non-admin in some devcluster and it worked for him.

Comment (matejvasek) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: @lance if you can reproduce the auth error please let me know how.

Comment (abrennan89) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: > By default the internal OpenShift registry is used, this behariour can be overridden by the -r flag (or FUNC_REGISTRY envvar).

I am very confused at this point; are we removing the `-r` flag or not?

Maybe we can remove this text entirely since we have [this section](https://docs.openshift.com/container-platform/4.10/serverless/functions/serverless-functions-setup.html#prerequisites_serverless-functions-setup) already? Surely if they follow the docs for setting up as we request them to at the beginning of the getting started, none of this should be an issue anyway?

Comment (abrennan89) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: >It's entirely likely that most OpenShift administrators haven't exposed the internal registry in this way - especially not for junior or mid level developers, who could easily be the kind of developers writing functions. If this is the case, and they try to deploy a function without the proper privileges, they will get an error. I think it might be important to add some cautionary text, e.g.

In that case I would recommend not even mentioning it as an option to be totally honest. That is a very big "IF" and I'm not sure why it would be useful to call it out as an option if realistically it's probably not an option for most devs?

I think in general we need more thought going into the E2E user stories for some of this set up stuff for functions. It feels like we keep running into issues when we come to the point of trying to figure out how to document something because we're not sure how it's supposed to work / what the recommended use cases are, or whether they can even be achieved by users in reality.

Comment (lance) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: @abrennan89 sorry to have caused a ruckus. Please disregard my comment. I think this PR looks good.

Comment (lance) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: @matejvasek comment is a nice improvement though.
```suggestion
This guide explains how you can get started with creating, building, and deploying a function on an {ServerlessProductName} installation. When building and deploying functions, the resulting container image is stored in an image registry. This can be a public registry, such as Quay or the OpenShift Container Registry. By default the OpenShift Container Registry is used, this behavior can be overridden using the `-r` flag or the `FUNC_REGISTRY` environment variable.
```

Comment (matejvasek) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: What Lance suggested above looks good.

The `-r` flag is also mentioned here: https://docs.openshift.com/container-platform/4.10/serverless/functions/serverless-functions-getting-started.html#serverless-build-func-kn_serverless-functions-getting-started
It states that is has to be set which is not true anymore.
We might remove notion of the `-r` flag from there since it is not mandatory.

Comment (matejvasek) [reply serverless/functions/serverless-functions-getting-started.adoc:9]:
```diff
diff --git a/modules/serverless-build-func-kn.adoc b/modules/serverless-build-func-kn.adoc
index 41fb395a2c..ae32d63596 100644
--- a/modules/serverless-build-func-kn.adoc
+++ b/modules/serverless-build-func-kn.adoc
@@ -8,13 +8,15 @@
 
 Before you can run a function, you must build the function project by using the `kn func build` command. The build command reads the `func.yaml` file from the function project directory to determine the image name and registry.
 
+By default the OpenShift Container Registry is used in `func.yaml` when creating new project. The registry used to store Function images can be overridden using the `-r` flag or the `FUNC_REGISTRY` environment variable.
+
 .Example `func.yaml`
 [source,yaml]
 ----
 name: example-function
 namespace: default
 runtime: node
-image: <image_from_registry>
+image: image-registry.openshift-image-registry.svc:5000/<oc-project>/<func-name>
 imageDigest: ""
 trigger: http
 builder: default
@@ -23,8 +25,6 @@ builderMap:
 envs: {}
 ----
 
-If the image name and registry are not set in the `func.yaml` file, you must either specify the registry flag, `-r` when using the `kn func build` command, or you are prompted to provide a registry value in the terminal when building a function. An image name is then derived from the registry value that you have provided.
-
 .Example command using the `-r` registry flag
 [source,terminal]
 ----
diff --git a/serverless/functions/serverless-functions-getting-started.adoc b/serverless/functions/serverless-functions-getting-started.adoc
index 135b281bf7..9bbad7e3cf 100644
--- a/serverless/functions/serverless-functions-getting-started.adoc
+++ b/serverless/functions/serverless-functions-getting-started.adoc
@@ -6,7 +6,7 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-This guide explains how you can get started with creating, building, and deploying a function on an {ServerlessProductName} installation. When building and deploying functions, the resulting container image is stored in an image registry. Usually this is a public registry, such as Quay. However, you can use the integrated OpenShift Container Registry instead if it has been exposed by a cluster administrator.
+This guide explains how you can get started with creating, building, and deploying a function on an {ServerlessProductName} installation.
 
 :FeatureName: {FunctionsProductName}
 include::snippets/technology-preview.adoc[leveloffset=+2]
 ```

File: modules/serverless-build-func-kn.adoc
Comment (lance) [inline modules/serverless-build-func-kn.adoc:9]: In the spirit of keeping func.yaml out of the examples as much as possible...
```suggestion
Before you can run a function, you must build the function project using the `kn func build` command. The build command uses the function project name and the image registry name to construct a fully qualified image name for your function.

.Example `func.yaml`
----
image: image-registry.openshift-image-registry.svc:5000/<oc_project>/<function_name>
----
```

File: modules/serverless-build-func-kn.adoc
Comment (lance) [inline modules/serverless-build-func-kn.adoc:25]: If you accept my above suggestion, then delete all of this.

File: modules/serverless-build-func-kn.adoc
Comment (lance) [inline modules/serverless-build-func-kn.adoc:26]:
```suggestion
By default, the OpenShift Container Registry is used as the image registry for storing function images. This can be overridden using the `-r` flag or the `FUNC_REGISTRY` environment variable.
```

File: modules/serverless-build-func-kn.adoc
Comment (lance) [inline modules/serverless-build-func-kn.adoc:31]: `func build` now has a lot of flags. I wonder if it makes sense to show any here.  I would suggest removing all flags from the example, or having some text such as "To learn more about the build command options, use the help command `kn func help build`".

File: modules/serverless-build-func-kn.adoc
Comment (lance) [inline modules/serverless-build-func-kn.adoc:37]:
```suggestion
$ kn func build -r quay.io/username
```
I would use a specific example that correlates with the output sample below.

File: modules/serverless-deploy-func-kn.adoc
Comment (lance) [inline modules/serverless-deploy-func-kn.adoc:9]: Do we use the term "service" generically like this? Or should it be the specific Knative `Service` primitive? Genuine question. E.g.
```suggestion
You can deploy a function to your cluster as a Knative `Service` by using the `kn func deploy` command. If the targeted function is already deployed, it is updated with a new container image that is pushed to a container image registry, and the Knative `Service` is updated.
```

Comment (abrennan89) [reply modules/serverless-deploy-func-kn.adoc:9]: We only use `Service` if we're referring to a specific "`Service` object". For general use it's "Knative service". Same sort of thing for any resources like this, e.g. general use is "config map", "channel", etc unless we're referring to a specific example of an object, e.g. "`ConfigMap` object", "`Channel` object", etc.. See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

Worth noting that upstream doc guidelines for this are different to downstream. I'd like to see it standardized, but it will not be changing for downstream, and there was push back on changing it upstream 🤷🏻

File: modules/serverless-build-func-kn.adoc
Comment (abrennan89) [reply modules/serverless-build-func-kn.adoc:9]: I'd remove "Example `func.yaml`" entirely then. I don't think just showing this one `image` line adds anything useful, really it just removes any context for users to see _how_ it fits into func.yaml.

**Review by matejvasek — APPROVED** [review]

**Review by drekar — APPROVED** [review]

**Review by lkingland — APPROVED** [review]
/lgtm as well

**Comment by abrennan89** [comment]
Thanks folks, will wait for the OK from @lance, QE, and OCP peer reviews before merging

**Comment by jrangelramos** [comment]
QE Approved

**Review by lance — COMMENTED** [review]
/lgtm

File: modules/serverless-build-func-kn.adoc
Comment (jc-berger) [inline modules/serverless-build-func-kn.adoc:9]: NBD but you don't need a comma before `or on a Kubernetes cluster.`

File: serverless/functions/serverless-functions-getting-started.adoc
Comment (jc-berger) [inline serverless/functions/serverless-functions-getting-started.adoc:9]: IBM SG wants us to avoid self referential language like "this guide." Can you simply say?:

`Get started with creating, building, and deploying a function on an {prod} installation.`

**Review by jc-berger — COMMENTED** [review]
I used the openshift docs peer review checklist when reviewing this PR: https://docs.google.com/document/d/1WN7k72PKxcPA__erp5TK-CefmS_sVwE3r40PvpmNk3E/edit#heading=h.mgz8v4w1kg86

File: serverless/functions/serverless-functions-getting-started.adoc
Comment (abrennan89) [reply serverless/functions/serverless-functions-getting-started.adoc:9]: This text wasn't added in this PR (just moved), and there is a follow up Jira related to this to fix the abstracts for all the Functions docs and make them IBM SG and Jupiter compliant (https://issues.redhat.com/browse/SRVCOM-1831) so I'll leave it as is for now. "Get started with creating, building, and deploying a function on an {prod} installation." doesn't really read properly to me.

File: modules/serverless-build-func-kn.adoc
Comment (abrennan89) [reply modules/serverless-build-func-kn.adoc]: Good catch, also updated Kubernetes -> {product-title} since these are product docs 😉

**Review by jc-berger — APPROVED** [review]

**Pull request merged**
